junifer/preprocess/confounds/fmriprep_confound_remover.py

```diff
@@ -528,100 +513,78 @@ def _validate_data(
-            include the ``BOLD_confounds`` key.
+            The other fields in the Junifer Data object. Must include the
+            ``BOLD_confounds`` key.
 
         Returns
         -------
-        Niimg-like object
-            Input image with confounds removed.
+        dict
+            The computed result as dictionary.
+        dict or None
+            If `self.masks` is not None, then the target data computed mask is
+            returned else None.
 
         """
-        assert extra_input is not None  # Not the case, data is validated
-        confounds_df = self._pick_confounds(extra_input["BOLD_confounds"])
-        confounds_array = confounds_df.values
-
+        # Validate data
+        self._validate_data(input, extra_input)
+        # Pick confounds
+        confounds_df = self._pick_confounds(extra_input["BOLD_confounds"])  # type: ignore
+        # Get BOLD data
         bold_img = input["data"]
+        # Set t_r
         t_r = self.t_r
         if t_r is None:
-            logger.info("No `t_r` specified, using t_r from nifti header")
-            zooms = bold_img.header.get_zooms()  # type: ignore
-            t_r = zooms[3]
+            logger.info("No `t_r` specified, using t_r from NIfTI header")
+            t_r = bold_img.header.get_zooms()[3]  # type: ignore
             logger.info(
-                f"Read t_r from nifti header: {t_r}",
+                f"Read t_r from NIfTI header: {t_r}",
             )
-
+        # Set mask data
         mask_img = None
+        bold_mask_dict = None
         if self.masks is not None:
             logger.debug(f"Masking with {self.masks}")
             mask_img = get_mask(
                 masks=self.masks, target_data=input, extra_input=extra_input
             )
-            # Save the mask in the extra input and link it to the bold data
+            # Return the BOLD mask and link it to the BOLD data type dict;
             # this allows to use "inherit" down the pipeline
             if extra_input is not None:
-                logger.debug("Setting mask_item")
-                extra_input["BOLD_mask"] = {
-                    "data": mask_img,
-                    "space": input["space"],
-                }
+                logger.debug("Setting `BOLD.mask_item`")
                 input["mask_item"] = "BOLD_mask"
-
-        logger.info("Cleaning image")
+                bold_mask_dict = {
+                    "BOLD_mask": {
+                        "data": mask_img,
+                        "space": input["space"],
+                    }
+                }
+        # Clean image
+        logger.info("Cleaning image using nilearn")
         logger.debug(f"\tdetrend: {self.detrend}")
         logger.debug(f"\tstandardize: {self.standardize}")
         logger.debug(f"\tlow_pass: {self.low_pass}")
         logger.debug(f"\thigh_pass: {self.high_pass}")
-        clean_bold = clean_img(
+        input["data"] = nimg.clean_img(
```

Comment:
I still do prefer to have it coded without explicit side-effects as it was before.

Reply:
I debated on this in my head for some time and decided to go with this due to the reduced complexity of having everything in `preprocess` and the need to now return `BOLD_mask` via the `extra_input` which is much more explicit in this.

Reply:
If you feel strongly about it and don't see an upside with the change, I can revert it with the required change.